[0.180s] Invoking command in '/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/src/ros2_aruco': AMENT_PREFIX_PATH=/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ros2_aruco:/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ros2_aruco_interfaces:/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ropislam:/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ropisim:/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ropinav:/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ropiapp:/opt/ros/humble PYTEST_ADDOPTS=--tb=short --junit-xml=/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/build/ros2_aruco/pytest.xml --junit-prefix=ros2_aruco -o cache_dir=/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/build/ros2_aruco/.pytest_cache PYTHONPATH=/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ros2_aruco/lib/python3.10/site-packages:/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ros2_aruco_interfaces/local/lib/python3.10/dist-packages:/opt/ros/humble/lib/python3.10/site-packages:/opt/ros/humble/local/lib/python3.10/dist-packages /usr/bin/python3 -m pytest
[0.314s] [1m============================= test session starts ==============================[0m
[0.314s] platform linux -- Python 3.10.12, pytest-6.2.5, py-1.10.0, pluggy-0.13.0
[0.316s] cachedir: /home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/build/ros2_aruco/.pytest_cache
[0.317s] rootdir: /home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/src/ros2_aruco
[0.317s] plugins: launch-testing-ros-0.19.6, launch-testing-1.0.4, ament-pep257-0.12.9, ament-flake8-0.12.9, ament-lint-0.12.9, ament-xmllint-0.12.9, ament-copyright-0.12.9, cov-3.0.0, colcon-core-0.15.1
[0.354s] [1mcollecting ... [0m[1m
[0.354s] collected 3 items                                                              [0m
[0.354s] 
[0.422s] test/test_copyright.py [31mF[0m[31m                                                 [ 33%][0m
[0.498s] test/test_flake8.py [31mF[0m[31m                                                    [ 66%][0m
[0.516s] test/test_pep257.py [31mF[0m[31m                                                    [100%][0m
[0.516s] 
[0.516s] =================================== FAILURES ===================================
[0.516s] [31m[1m________________________________ test_copyright ________________________________[0m
[0.516s] [1m[31mtest/test_copyright.py[0m:23: in test_copyright
[0.535s]     [94massert[39;49;00m rc == [94m0[39;49;00m, [33m'[39;49;00m[33mFound errors[39;49;00m[33m'[39;49;00m
[0.535s] [1m[31mE   AssertionError: Found errors[0m
[0.535s] [1m[31mE   assert 1 == 0[0m
[0.535s] ----------------------------- Captured stderr call -----------------------------
[0.535s] launch/aruco_recognition.launch.py: could not find copyright notice
[0.535s] ros2_aruco/aruco_generate_marker.py: could not find copyright notice
[0.535s] ros2_aruco/aruco_node.py: could not find copyright notice
[0.536s] 3 errors, checked 7 files
[0.536s] [31m[1m_________________________________ test_flake8 __________________________________[0m
[0.536s] [1m[31mtest/test_flake8.py[0m:23: in test_flake8
[0.536s]     [94massert[39;49;00m rc == [94m0[39;49;00m, [33m'[39;49;00m[33mFound errors[39;49;00m[33m'[39;49;00m
[0.536s] [33m=============================== warnings summary ===============================[0m
[0.536s] test/test_flake8.py::test_flake8
[0.536s] test/test_flake8.py::test_flake8
[0.536s]   Warning: SelectableGroups dict interface is deprecated. Use select.
[0.536s] 
[0.536s] -- Docs: https://docs.pytest.org/en/stable/warnings.html
[0.536s] [1m[31mE   AssertionError: Found errors[0m
[0.536s] [1m[31mE   assert 1 == 0[0m
[0.536s] ----------------------------- Captured stdout call -----------------------------
[0.536s] 
[0.536s] ./launch/aruco_recognition.launch.py:6:1: E302 expected 2 blank lines, found 1
[0.536s] def generate_launch_description():
[0.536s] ^
[0.536s] 
[0.536s] ./ros2_aruco/aruco_generate_marker.py:15:1: D204 1 blank line required after class docstring
[0.536s]     """ Trick to allow both defaults and nice formatting in the help. """
[0.536s] ^
[0.536s] 
[0.536s] ./ros2_aruco/aruco_generate_marker.py:15:1: D210 No whitespaces allowed surrounding docstring text
[0.536s]     """ Trick to allow both defaults and nice formatting in the help. """
[0.536s] ^
[0.536s] 
[0.536s] ./ros2_aruco/aruco_node.py:121:36: E721 do not compare types, use 'isinstance()'
[0.536s]             if type(dictionary_id) != type(cv2.aruco.DICT_5X5_100):
[0.536s]                                    ^
[0.536s] 
[0.536s] 1     D204 1 blank line required after class docstring
[0.536s] 1     D210 No whitespaces allowed surrounding docstring text
[0.536s] 1     E302 expected 2 blank lines, found 1
[0.537s] 1     E721 do not compare types, use 'isinstance()'
[0.537s] 
[0.537s] 8 files checked
[0.537s] 4 errors
[0.537s] 
[0.537s] 'D'-type errors: 2
[0.537s] 'E'-type errors: 2
[0.537s] 
[0.537s] Checked files:
[0.537s] 
[0.537s] * ./setup.py
[0.537s] * ./launch/aruco_recognition.launch.py
[0.537s] * ./test/test_pep257.py
[0.537s] * ./test/test_flake8.py
[0.537s] * ./test/test_copyright.py
[0.537s] * ./ros2_aruco/aruco_generate_marker.py
[0.537s] * ./ros2_aruco/__init__.py
[0.537s] * ./ros2_aruco/aruco_node.py
[0.537s] ----------------------------- Captured stderr call -----------------------------
[0.537s] option --max-complexity: please update from optparse string `type=` to argparse callable `type=` -- this will be an error in the future
[0.537s] [31m[1m_________________________________ test_pep257 __________________________________[0m
[0.537s] [1m[31mtest/test_pep257.py[0m:23: in test_pep257
[0.537s]     [94massert[39;49;00m rc == [94m0[39;49;00m, [33m'[39;49;00m[33mFound code style errors / warnings[39;49;00m[33m'[39;49;00m
[0.537s] [1m[31mE   AssertionError: Found code style errors / warnings[0m
[0.537s] [1m[31mE   assert 1 == 0[0m
[0.537s] ----------------------------- Captured stdout call -----------------------------
[0.537s] checking ./launch/aruco_recognition.launch.py
[0.537s] checking ./ros2_aruco/__init__.py
[0.537s] checking ./ros2_aruco/aruco_generate_marker.py
[0.537s] ./ros2_aruco/aruco_generate_marker.py:15 in public class `CustomFormatter`: D204: 1 blank line required after class docstring (found 0)
[0.537s] ./ros2_aruco/aruco_generate_marker.py:15 in public class `CustomFormatter`: D210: No whitespaces allowed surrounding docstring text
[0.537s] checking ./ros2_aruco/aruco_node.py
[0.537s] ./ros2_aruco/aruco_node.py:1 at module level: D407: Missing dashed underline after section ('Parameters')
[0.537s] ./ros2_aruco/aruco_node.py:1 at module level: D406: Section name should end with a newline ('Parameters', not 'Parameters:')
[0.537s] checking ./setup.py
[0.537s] checking ./test/test_copyright.py
[0.537s] checking ./test/test_flake8.py
[0.537s] checking ./test/test_pep257.py
[0.537s] checking test/test_copyright.py
[0.537s] checking test/test_flake8.py
[0.537s] checking test/test_pep257.py
[0.537s] 4 errors
[0.537s] - generated xml file: /home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/build/ros2_aruco/pytest.xml -
[0.537s] =========================== short test summary info ============================
[0.538s] FAILED test/test_copyright.py::test_copyright - AssertionError: Found errors
[0.538s] FAILED test/test_flake8.py::test_flake8 - AssertionError: Found errors
[0.538s] FAILED test/test_pep257.py::test_pep257 - AssertionError: Found code style er...
[0.538s] [31m======================== [31m[1m3 failed[0m, [33m2 warnings[0m[31m in 0.22s[0m[31m =========================[0m
[0.569s] Invoked command in '/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/src/ros2_aruco' returned '1': AMENT_PREFIX_PATH=/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ros2_aruco:/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ros2_aruco_interfaces:/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ropislam:/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ropisim:/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ropinav:/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ropiapp:/opt/ros/humble PYTEST_ADDOPTS=--tb=short --junit-xml=/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/build/ros2_aruco/pytest.xml --junit-prefix=ros2_aruco -o cache_dir=/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/build/ros2_aruco/.pytest_cache PYTHONPATH=/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ros2_aruco/lib/python3.10/site-packages:/home/tej/Documents/CNH/DockerImages/glxgears/code/warehouse_autonomous_robo_picker/install/ros2_aruco_interfaces/local/lib/python3.10/dist-packages:/opt/ros/humble/lib/python3.10/site-packages:/opt/ros/humble/local/lib/python3.10/dist-packages /usr/bin/python3 -m pytest
